cmd/operator/internal/controller/sinkfilter_controller.go

```diff
@@ -0,0 +1,454 @@
+				if backoff > maxBackoff {
+					backoff = maxBackoff
+				}
+				continue
+			case <-watchInfo.StopCh:
+				return
+			}
+		}
+
+		backoff = time.Second // Reset backoff on successful connection
+
+		log.Info("Started watch", "key", key, "gvr", watchInfo.GVR.String(), "resourceVersion", watchInfo.ResourceVersion)
+
+		r.processWatchEvents(ctx, watchInfo.WatchInterface, watchInfo, key)
+
+		watchInfo.WatchInterface.Stop()
+		watchInfo.WatchInterface = nil
+
+		log.Info("Watch disconnected, will retry", "key", key)
+	}
+}
+
+func (r *SinkFilterReconciler) createWatch(ctx context.Context, gvr schema.GroupVersionResource, resourceVersion string) (watch.Interface, error) {
+	listOptions := metav1.ListOptions{
+		TimeoutSeconds:  func() *int64 { t := int64(300); return &t }(), // 5 minute timeout
+		ResourceVersion: resourceVersion,
+	}
+
+	watchInterface, err := r.dynamicClient.Resource(gvr).Watch(ctx, listOptions)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create watch for %v: %w", gvr, err)
+	}
+	return watchInterface, nil
+}
+
+func (r *SinkFilterReconciler) processWatchEvents(ctx context.Context, watchInterface watch.Interface, watchInfo *WatchInfo, key string) {
+	log := log.FromContext(ctx)
+	resultChan := watchInterface.ResultChan()
+
+	for {
+		select {
+		case <-watchInfo.StopCh:
+			log.Info("Stopping watch event processing", "key", key)
+			return
+		case event, ok := <-resultChan:
+			if !ok {
+				log.Info("Watch channel closed", "key", key)
+				return
+			}
+
+			// If watch event type is ERROR, exit the loop to recreate the watch
+			if event.Type == watch.Error {
+				// Handle specific error codes
+				if status, ok := event.Object.(*metav1.Status); ok && status.Reason == metav1.StatusReasonGone {
+					// Gone - resource version is too old, clear it for full resync
+					log.Info("Watch received Gone error, clearing resource version for full resync", "key", key)
+					watchInfo.ResourceVersion = ""
+				}
+
+				return
+			}
+
+			// Update resource version from the event object
+			if unstructuredObj, ok := event.Object.(*unstructured.Unstructured); ok {
+				if resourceVersion := unstructuredObj.GetResourceVersion(); resourceVersion != "" {
+					watchInfo.ResourceVersion = resourceVersion
+				}
+			}
+
+			if err := r.handleWatchEvent(ctx, event, watchInfo); err != nil {
+				log.Error(err, "Failed to handle watch event", "key", key)
+			}
+		}
+	}
+}
+
+func (r *SinkFilterReconciler) handleWatchEvent(ctx context.Context, event watch.Event, watchInfo *WatchInfo) error {
+	unstructuredObj, ok := event.Object.(*unstructured.Unstructured)
+	if !ok {
+		return fmt.Errorf("unexpected object type: %T", event.Object)
+	}
+
+	objNamespace := unstructuredObj.GetNamespace()
+	_, globalExists := watchInfo.Namespaces[constants.SinkFilterGlobalNamespace]
+	_, namespaceExists := watchInfo.Namespaces[objNamespace]
+
+	if !globalExists && !namespaceExists {
+		return nil // Skip this event
+	}
```

Comment:
This keeps sending messages of namespaces we don't track (they not are instrumented), for example:

```
time=2025-10-07T13:30:14.852Z level=INFO source=github.com/kubearchive/kubearchive/cmd/operator/internal/controller/sinkfilter_controller.go:433 msg="Cloud event sent successfully" controller=sinkfilter controllerGroup=kubearchive.org controllerKind=SinkFilter SinkFilter.name=sink-filters SinkFilter.namespace=product-kubearchive namespace=product-kubearchive name=sink-filters reconcileID=7efa01ec-2baf-4bcc-aae9-bf0cb8d7657e eventType=update gvr="/v1, Resource=pods" kind=Pod name=redhat-operators-dhs8t namespace=openshift-marketplace
```

I'm fine with handling this on a next PR.